spyder/utils/icon_manager.py

```diff
@@ -366,3 +384,56 @@ def icon(name, resample=False, icon_path=None):
             if osp.isfile(icon_path):
                 icon = QIcon(icon_path)
         return icon if icon is not None else QIcon()
+
+
+def get_icon_by_extension(fname):
```

Comment:
I like this refactoring!! It was a really good idea to have all of this grouped under a single function.

Nicely done @juanis2112! (or was it @dalthviz idea?)

Reply:
I would say it was a team effort :+1:

Reply:
Great to hear!